### PR TITLE
Phase2-hgx106 Modify the HGCal geometry to introduce z-dependent wafer type

### PR DESCRIPTION
**Pull request by bsunanda**
More realistic description of wafer structure

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc [hunk 1]
```diff
@@ -116,6 +125,10 @@ void DDHGCalEEAlgo::initialize(const DDNumericArguments & nArgs,
 #ifdef EDM_ML_DEBUG
   edm::LogVerbatim("HGCalGeom") << "DDHGCalEEAlgo: NameSpace " << nameSpace_;
 #endif
+
+  waferType_ = new HGCalWaferType(rad100to200_, rad200to300_,
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc]: could this be a smart pointer?

File: Geometry/HGCalCommonData/src/HGCalWaferType.cc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+HGCalWaferType::HGCalWaferType(const std::vector<double>& rad100, 
+			       const std::vector<double>& rad200,
+			       double waferSize, double zMin, 
+			       int cutValue) : rad100_(rad100),
+					       rad200_(rad200),
+					       waferSize_(waferSize),
+					       zMin_(zMin),
+					       cutValue_(cutValue) {
+  r_ = 0.5*waferSize_;
+  R_ = waferSize_/std::sqrt(3.0);
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalWaferType.cc]: define `sqrt(3.0)` as a const

**Comment by davidlange6** [comment]
> On Mar 23, 2018, at 4:47 PM, Kevin Pedro <notifications@github.com> wrote:
> 
> @kpedro88 commented on this pull request.
> 
> In Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc:
> 
> > @@ -116,6 +125,10 @@ void DDHGCalEEAlgo::initialize(const DDNumericArguments & nArgs,
>  #ifdef EDM_ML_DEBUG
>    edm::LogVerbatim("HGCalGeom") << "DDHGCalEEAlgo: NameSpace " << nameSpace_;
>  #endif
> +
> +  waferType_ = new HGCalWaferType(rad100to200_, rad200to300_,
> 
> could this be a smart pointer?
> 
> In Geometry/HGCalCommonData/src/HGCalWaferType.cc:
> 
> > +#include<cmath>
> +
> +//#define EDM_ML_DEBUG
> +
> +const double k_ScaleFromDDD = 0.1;
> +
> +HGCalWaferType::HGCalWaferType(const std::vector<double>& rad100, 
> +			       const std::vector<double>& rad200,
> +			       double waferSize, double zMin, 
> +			       int cutValue) : rad100_(rad100),
> +					       rad200_(rad200),
> +					       waferSize_(waferSize),
> +					       zMin_(zMin),
> +					       cutValue_(cutValue) {
> +  r_ = 0.5*waferSize_;
> +  R_ = waferSize_/std::sqrt(3.0);
> 
> define sqrt(3.0) as a const

even better - 1.0/sqrt(3.)

File: Geometry/HGCalCommonData/plugins/DDHGCalHEAlgo.cc [hunk 3]
```diff
@@ -449,10 +459,10 @@ void DDHGCalHEAlgo::positionMix(const DDLogicalPart& glog,
...
 				  << " with " << rot;
 #endif
+    if (layerSenseBot_[ly] != 0)
+      positionSensitive(glog2,rin,rmid,zz+zpos,layerSenseBot_[ly],cpv);
     zpos += hthickl;
     ++copyNumberBot_[ii];
-    if (layerSenseBot_[ly] != 0)
-      positionSensitive(glog2,rin,rmid,layerSenseBot_[ly],cpv);
   }
   if (fabs(thickTot-thick) < 0.00001) {
```
Comment (civanch) [inline Geometry/HGCalCommonData/plugins/DDHGCalHEAlgo.cc]: @bsunanda , if variables are double, then you do not need fabs

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**